src/core/symbology/classification/qgsclassificationmethod.h

```diff
@@ -0,0 +1,290 @@
+    QgsClassificationRange( const QString &label, const double &lowerBound, const double &upperBound )
+      : mLabel( label )
+      , mLowerBound( lowerBound )
+      , mUpperBound( upperBound )
+    {}
+    //! Returns the lower bound
+    double lowerBound() const {return mLowerBound;}
+    //! Returns the upper bound
+    double upperBound() const {return mUpperBound;}
+
+    //! Returns the lower bound
+    QString label() const {return mLabel;}
+
+  private:
+    QString mLabel;
+    double mLowerBound;
+    double mUpperBound;
+};
+
+
+
+/**
+ * \ingroup core
+ * QgsClassification is an abstract class for implementations of classification methods
+ * \since QGIS 3.10
+ */
+class CORE_EXPORT QgsClassificationMethod SIP_ABSTRACT
+{
+
+#ifdef SIP_RUN
+    SIP_CONVERT_TO_SUBCLASS_CODE
+    if ( dynamic_cast<QgsClassificationEqualInterval *>( sipCpp ) )
+      sipType = sipType_QgsClassificationEqualInterval;
+    else if ( dynamic_cast<QgsClassificationJenks *>( sipCpp ) )
+      sipType = sipType_QgsClassificationJenks;
+    else if ( dynamic_cast<QgsClassificationPrettyBreaks *>( sipCpp ) )
+      sipType = sipType_QgsClassificationPrettyBreaks;
+    else if ( dynamic_cast<QgsClassificationQuantile *>( sipCpp ) )
+      sipType = sipType_QgsClassificationQuantile;
+    else if ( dynamic_cast<QgsClassificationStandardDeviation *>( sipCpp ) )
+      sipType = sipType_QgsClassificationStandardDeviation;
+    else
+      sipType = 0;
+    SIP_END
+#endif
+
+  public:
+
+    //! Defines the class position
+    enum ClassPosition
+    {
+      LowerBound,
+      Inner,
+      UpperBound
+    };
+
+    /**
+      * Creates a classification method.
+      * \param valuesRequired if TRUE, this means that the method requires a set of values to determine the classes
+      * \param symmetricModeAvailable if TRUE, this allows using symmetric classification
+      * \param codeCommplexity as the exponent in the big O notation
+      * \param
+      */
+    explicit QgsClassificationMethod( bool valuesRequired, bool symmetricModeAvailable, int codeComplexity = 1 );
+
+    virtual ~QgsClassificationMethod() = default;
+
+    /**
+     * Returns a clone of the method.
+     * Implementation can take advantage of copyBase method which copies the parameters of the base class
+     */
+    virtual QgsClassificationMethod *clone() const = 0 SIP_FACTORY;
+
+    //! The readable and translate name of the method
+    virtual QString name() const = 0;
+
+    //! The id of the method
+    virtual QString id() const = 0; // as saved in the project, must be unique in registry
+
+    /**
+     * Returns the label for a range
+     */
+    virtual QString labelForRange( const double &lowerValue, const double &upperValue, ClassPosition position = Inner ) const;
+
+
+    //! Writes extra information about the method
+    virtual void saveExtra( QDomElement &element, const QgsReadWriteContext &context ) const {Q_UNUSED( element ); Q_UNUSED( context )}
```

Comment:
probably better to stick with the quasi-standard readXml/writeXml here

Reply:
I already have create/save. I wanted to clearly state it's for additional parameters. Is this obvious?